extensions/stackdriver/metric/registry.cc

```diff
@@ -201,66 +201,86 @@ StackdriverOptions getStackdriverOptions(
 /*
  *  view function macros
  */
-#define REGISTER_COUNT_VIEW(_v)                              \
-  void register##_v##View(absl::Duration expiry_duration) {  \
-    const ViewDescriptor view_descriptor =                   \
-        ViewDescriptor()                                     \
-            .set_name(k##_v##View)                           \
-            .set_measure(k##_v##Measure)                     \
-            .set_expiry_duration(expiry_duration)            \
-            .set_aggregation(Aggregation::Count()) ADD_TAGS; \
-    View view(view_descriptor);                              \
-    view_descriptor.RegisterForExport();                     \
+#define REGISTER_COUNT_VIEW(_v)                                            \
+  void register##_v##View(absl::Duration expiry_duration,                  \
+                          std::vector<std::string_view> dropped_metrics) { \
+    auto iter = std::find(dropped_metrics.begin(), dropped_metrics.end(),  \
+                          k##_v##View);                                    \
+    if (iter != dropped_metrics.end()) return;                             \
+    const ViewDescriptor view_descriptor =                                 \
+        ViewDescriptor()                                                   \
+            .set_name(k##_v##View)                                         \
+            .set_measure(k##_v##Measure)                                   \
+            .set_expiry_duration(expiry_duration)                          \
+            .set_aggregation(Aggregation::Count()) ADD_TAGS;               \
+    View view(view_descriptor);                                            \
+    view_descriptor.RegisterForExport();                                   \
```

Comment:
I don't think this will work if user configure dropping metric in the middle of proxy lifetime. Opencensus maintains everything (measurement, views) with a singleton, skip registering won't revert previous registered view. User will need to restart proxy to make the config effective. Not sure if Opencensus provides a way to deregister or not (I don't think so..), so probably need to document this clearly.

Reply:
This is a good point. I'll look into deregistration.

Reply:
added deregistration support.